src/gui/dialogs/story_viewer.hpp

```diff
@@ -75,6 +75,11 @@ class story_viewer : public modal_dialog
 
 	void flag_stack_as_dirty();
 
+	// Manually close the dialog, since widget with id=cancel is not a button
```

Comment:
Why is it not a button? Would it not be easier if it were a button?

Reply:
A button would not fit with the overall design, hence the work around.

Reply:
If we want to make it a button, we need some kind of ornate button, which would also require some more assets.

Reply:
I think you're missing the point? A button can have any appearance you want. You can make a button that looks exactly like your skip text. (You'd probably want a different hovered appearance as well though.)

Reply:
Right, can't believe I missed that. And the hover appearance change would be a nice addition. Will try to do that.

Reply:
Now everything is a button (though I don't like it that canvas doesn't have relative positioning.) and it also makes the code a bit more cleaner.